rdwatch/core/tasks/animation_export.py

```diff
@@ -746,10 +746,12 @@ def create_modelrun_animation_export(
     site_tasks = [
         create_site_animation_export.s(site_id, settings, userId)
         for site_id in SiteImage.objects.filter(site__configuration_id=modelrun_id)
-        .values_list('site_id', flat=True)
-        .iterator()
+        .values('site_id')  # Group by site_id
+        .annotate(image_count=Count('id'))  # Count images per site_id
```

Comment:
I haven't tested yet, but shouldn't this be

```suggestion
        .annotate(image_count=Count('site_id'))  # Count images per site_id
```

?